### PR TITLE
support scaled_dot_product_attention for shifted window attention

### PR DESCRIPTION
**Pull request by yokosyun**
# Changes
- support torch.nn.functional.scaled_dot_product_attention for shifted_window_attention in swin.
- make the mask generation process as function

https://github.com/pytorch/vision/issues/8182

# Compatibility Test

```
    import torch
    import torchvision
    batch_size = 2
    height = 224
    width = 224
    img = torch.rand([batch_size, 3, height, width])
    model = torchvision.models.swin_t()
    ouput = model(img)

    model = torchvision.models.swin_t()
    ouput = model(img)

    model = torchvision.models.swin_v2_t()
    ouput = model(img)
```

# Unit Test
- got the identical result with current implementation, when enable_mem_efficient=False
```
    import torch
    from torchvision.models.swin_transformer import shifted_window_attention

    batch_size = 2
    channels = 96
    height = 128
    width = 128
    window_size = [7, 7]
    num_heads = 3
    shift_size = [0, 0]

    input = torch.rand([batch_size, height, width, channels])
    qkv_weight = torch.rand([3 * channels, channels])
    proj_weight = torch.rand([channels, channels])
    relative_position_bias = torch.rand(
        [1, num_heads, window_size[0] * window_size[1], window_size[0] * window_size[1]]
    )

    old_attn = shifted_window_attention(
        input=input,
        qkv_weight=qkv_weight,
        proj_weight=proj_weight,
        relative_position_bias=relative_position_bias,
        window_size=window_size,
        num_heads=num_heads,
        shift_size=shift_size,
        attention_dropout=0.0,
        dropout=0.0,
        qkv_bias=None,
        proj_bias=None,
        logit_scale=None,
        training=True,
        use_efficient_attention=False,
    )

    new_attn = shifted_window_attention(
        input=input,
        qkv_weight=qkv_weight,
        proj_weight=proj_weight,
        relative_position_bias=relative_position_bias,
        window_size=window_size,
        num_heads=num_heads,
        shift_size=shift_size,
        attention_dropout=0.0,
        dropout=0.0,
        qkv_bias=None,
        proj_bias=None,
        logit_scale=None,
        training=True,
        use_efficient_attention=True,
    )

    diff_attn = torch.abs(old_attn - new_attn)
    print(torch.sum(diff_attn))
```

### CODE REVIEW

**Comment by yokosyun** [comment]
attn_mask of scaled_dot_product_attention in torch doesn't support tensor type yet.
